### ISSUE TITLE
Remove Bootstrap CSS

### ISSUE THREAD

**Issue by koistya** [kriasoft/react-starter-kit#43]
What do you think about removing Bootstrap CSS from this template? In favor of some custom minimum basic styles.

**Comment by ChaosD** [comment]
i toyed with http://purecss.io/ for some time.
@koistya, are you planning to turn this in a yo generator? Otherwise, beside from being a lot of additional work for you anyways, supporting multiple frameworks might get quite messy.

**Comment by yemi** [comment]
The less opinionated the better. I would go for custom minimum basic styles since people most likely will use their css lib/framework/style of choice anyways.

**Comment by codedsignal** [comment]
The more dependencies on 3rd party libraries you include the harder it is to understand the structure and intent of the code. +1 on getting rid of Bootstrap in favour of a minimal custom stylesheet.

**Comment by koistya** [comment]
**Spoiler**: The CSS will follow this naming convention:

```css
.ComponentName { }
.ComponentName--modifier { }
.ComponentName-elementName { }
.ComponentName-elementName--modifier { }
```

Which is well suited for modular architecture. No shared styles or images whatsoever. Only styles per component. Btw, the same CSS naming convention is currently used at Twitter.

**Comment by codedsignal** [comment]
@koistya going BEM strikes me as a good move and completely appropriate to React's strength in creating modular components

**Comment by sdiaz** [comment]
It would be great, and what about BEM CSSO tool?

**Comment by koistya** [comment]
@sdiaz, I'm not sure about [CSSO](https://bem.info/tools/optimizers/csso/), I will check it. As of now, the starter kit is using [csscomb](http://csscomb.com/) and [clean-css](https://github.com/jakubpawlowicz/clean-css). Does CSSO have any advantages over these two?

**Comment by radum** [comment]
I agree with @yemi just remove it and people will use whatever they want.

**Comment by seemsindie** [comment]
I agree that it's should be  less opinionated, let the people use whatever they want.

**Comment by goatslacker** [comment]
Bootstrap is good enough to provide as a base. FWIW this should all be easily pluggable.

**Comment by ivanoats** [comment]
:+1: on making it easy to include other CSS frameworks like node-bourbon https://www.npmjs.com/package/node-bourbon

**Comment by joybro** [comment]
-1 for the following reasons.

1. It is very useful for "starters" like me to have bootstrap as default.
2. It would be not that hard for "experienced people" to remove or replace bootstrap.

**Comment by koistya** [comment]
A quote, from Web Starter Kit documentation, just replaced "Web Starter Kit" with "React Starter Kit":

> React Starter Kit doesn't aim to compete with CSS libraries like Bootstrap, Foundation and Pure. These libraries provide an excellent solution for prototyping your initial project. The biggest challenge they present is it’s almost too easy to get stuck using their look and feel for the lifetime of your site. We think this leads to a poorer experience on the multi-screen web.

> React Starter Kit provides boilerplate styles & a visual style guide for projects, but encourages customising these to fit your own site. This may need a little more work, but the reality is that any serious project is going to have its own look and feel. We want you to feel comfortable changing the kit to suit your own needs.

> If you wish to use Bootstrap or other CSS libraries in your React Starter Kit project, you have the flexibility to do so.

Bootstrap is removed in c92ea26d9ff9f377335aad5c9acad320dda32a51

**Comment by kpnigalye** [comment]
Can anybody help me with integrating Zurb foundation with react starter kit?

I have installed foundation using 'npm install foundation'. I am not sure if this is correct.

**Comment by koistya** [comment]
@kpnigalye there easiest integration approach is to reference Zurb Foundation from a public CDN:

https://cdnjs.com/libraries/foundation

Just edit `components/Html` to include appropriate `<link ../>` tags. Do you think this approach will work for you?

**Comment by ddewaele** [comment]
Started looking into the react-starter kit since a couple of days and really impressed. 

Also having some difficulties adding the bootstrap stylesheet. I tried the following.

- App.scss 

Added @import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css'  but got a failure during npm start.

- Html component render function 

Adding a link tag in the Html component. 

```<link rel="stylesheet" href="../../../node_modules/bootstrap/dist/css/bootstrap.min.css"/>```

But it doesn't pickup the css file as it is not present in the build folder

- Created public/css folder

Manually copied a bootstrap.css to the public/css folder and modified the copy.js script to also copy the css folder to the build folder.

The ```public/css``` folder thing works but feels like a hack. What would be the best way to add the bootstrap stylesheet ?
